Automattic/jetpack
# VideoPress: sync video `post_id` with block attribute `id`

This PR allows editing a VideoPress video block in that the video has been defined via a VideoPress URL, as long as the user is the video's owner.
Adding videos that the user is not the owner is a task that will be addressed in follow-ups.

Related to https://github.com/Automattic/jetpack/issues/27863

#### Changes proposed in this Pull Request:

* VideoPress: sync video `post_id` with block attribute `id`

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Ensure you already have a VideoPress video ready to use. URL shape: `https://videopress.com/v/<GUID>`. 
* If it's private, better:

<img width="672" alt="image" src="https://user-images.githubusercontent.com/77539/206852003-923d9f1a-8b08-4485-9a44-71eb3bb37221.png">

* Go to block editor
* Enable debug()

```
localStorage.setItem( 'debug', 'videopress:*' )
```

* Create a new video block either from the embed block or by setting the URL.

`v6` | `embed`
------|------
<img src="https://user-images.githubusercontent.com/77539/206851493-a496ca04-746e-4333-a4bc-2cb2b74563f6.png" /> | <img src="https://user-images.githubusercontent.com/77539/206851513-50d9c449-4d38-4aa4-ba2e-6feb70a4a216.png" />

* For the embed block, you should convert it to `v6`.

<img width="655" alt="image" src="https://user-images.githubusercontent.com/77539/206852055-e7892d6c-e4f8-46cd-a633-6cc2abf378cf.png">

* Confirm the app starts to sync the video ID
* Confirm you can edit the block attributes such as `title`, `description`, etc...

<img width="986" alt="Screen Shot 2022-12-10 at 11 10 19" src="https://user-images.githubusercontent.com/77539/206852104-6977498b-7925-498c-8054-20fdfba0c5a3.png">
